File: 3314-construct-the-minimum-bitwise-array-i/python3/3314-construct-the-minimum-bitwise-array-i.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        ans = [-1] * len(nums)
        for i, v in enumerate(nums):
            if v % 2 == 1:
                ans[i] = int(v - ((v+1) & (-v -1)) / 2)
        return ans

File: 3314-construct-the-minimum-bitwise-array-i/java/3314-construct-the-minimum-bitwise-array-i.java
class Solution {
  public int[] minBitwiseArray(List<Integer> nums) {
    int[] ans = new int[nums.size()];

    for (var i = 0; i < nums.size(); ++i) {
      for (var x = 0; x < nums.get(i); ++x) {
        if ((x | (x + 1)) == nums.get(i)) {
          ans[i] = x;
          break;
        } else {
          ans[i] = -1;
        }
      }
    }

    return ans;
  }
}
